web/modules/custom/dbcdk_community/src/Content/FlaggableContentRepository.php

```diff
@@ -182,4 +204,47 @@ public function getContentByIdAllFlags($flag_id) {
     return $flagged_content;
   }
 
+  /**
+   * Get collections for a FlaggableContent object.
+   *
+   * This method provides a simple way to fetch collections for a
+   * FlaggableContent object.
+   *
+   * @param \Drupal\dbcdk_community\Content\FlaggableContent $content
+   *   The FlaggableContent object we wish to fetch collections from.
+   * @param string $type
+   *   The type of collection we wish to fetch.
+   *
+   * @return \DBCDK\CommunityServices\Model\ImageCollection[]|\DBCDK\CommunityServices\Model\VideoCollection[]
+   *   On array of collections for the provided FlaggableContent object.
+   */
+  public function getCollections(FlaggableContent $content, $type = 'image') {
+    $object = $content->getObject();
+    // Instantiate a reflection of the class to find out what type of object
+    // we're fetching collections for.
+    $object_reflection = new \ReflectionClass($object);
+    $object_name = $object_reflection->getShortName();
+
+    // Get the API corresponding to the type of content (ex: $this->postApi).
+    $api = $this->{lcfirst($object_name) . 'Api'};
+    // Format the method we wish to use when fetching collections from the API
+    // defined above (ex: PostApi::postPrototypeGetImage()).
+    $collection_method = lcfirst($object_name) . 'PrototypeGet' . ucfirst(strtolower($type));
+    if (method_exists($api, $collection_method)) {
+      // If there's no collections for the ID, then the service will return a
+      // 404 that will cause a fatal error if we don't handle this with a "try".
+      try {
+        return $api->{$collection_method}($object->getId());
+      }
+      catch (\Exception $e) {
+        // We don't want to log the exception because this happens for every
+        // post or comment without images or videos.
+        return [];
```

Comment:
Silently swallowing the 404 exceptions is fine here, but I think we should consider rethrowing or at least logging ApiExceptions.